src/main/java/com/sap/oss/phosphor/fosstars/data/artifact/VulnerabilitiesFromNpmAudit.java

```diff
@@ -0,0 +1,242 @@
+import com.sap.oss.phosphor.fosstars.nvd.NVD;
+import com.sap.oss.phosphor.fosstars.nvd.data.NvdEntry;
+import com.sap.oss.phosphor.fosstars.util.Json;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.http.HttpHeaders;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+/**
+ * This data provider tries to fill out the {@link OssFeatures#VULNERABILITIES_IN_ARTIFACT}. It
+ * gathers vulnerabilities from {@link NpmArtifact} using NPM Audit.
+ *
+ * @see <a href="https://docs.npmjs.com/auditing-package-dependencies-for-security-vulnerabilities">NPM
+ * Audit</a>
+ */
+public class VulnerabilitiesFromNpmAudit implements DataProvider {
+
+  /**
+   * A logger.
+   */
+  private final Logger logger = LogManager.getLogger(getClass());
+
+  /**
+   * NPM registry URL to access security audit.
+   */
+  private static final String NPM_AUDIT_URL =
+      "https://registry.npmjs.org/-/npm/v1/security/audits";
+
+  /**
+   * NPM Audit API POST request body template.
+   */
+  private static final String
+      NPM_AUDIT_URL_BODY_ENTITY =
+      String.join("",
+          "{",
+          "\"name\": \"npm-vulnerabilities\",",
+          "\"version\": \"1.0.0\",",
+          "\"requires\" :{\"%s\": \"%s\"},",
+          "\"dependencies\": {\"%s\": {\"version\": \"%s\"}}",
+          "}");
+
+  /**
+   * An interface to NVD.
+   */
+  private final NVD nvd;
+
+  /**
+   * Initializes a data provider.
+   *
+   * @param nvd An {@link NVD}.
+   */
+  public VulnerabilitiesFromNpmAudit(NVD nvd) {
+    this.nvd = nvd;
+  }
+
+  /**
+   * The method always returns false, so that all child classes can't be interactive.
+   */
+  @Override
+  public final boolean interactive() {
+    return false;
+  }
+
+  /**
+   * This is a dummy cache which stores nothing.
+   */
+  @Override
+  public ValueCache<Subject> cache() {
+    return NoValueCache.create();
+  }
+
+  /**
+   * There is no call back required for this data provider.
+   */
+  @Override
+  public VulnerabilitiesFromNpmAudit set(UserCallback callback) {
+    return this;
+  }
+
+  /**
+   * No cache value is needed that is used by the data provider.
+   */
+  @Override
+  public VulnerabilitiesFromNpmAudit set(ValueCache<Subject> cache) {
+    return this;
+  }
+
+  /**
+   * No configuration is required for this data provider.
+   */
+  @Override
+  public VulnerabilitiesFromNpmAudit configure(Path config) throws IOException {
```

Comment:
```suggestion
  public VulnerabilitiesFromNpmAudit configure(Path config) {
```

The exception can be removed here.